src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,135 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+/**
+ * class for deadlines, a type of task, with an added deadline field
+ */
+public class Deadline extends Task {
+
+    private String deadline;
+    private LocalDate deadlineDate = null;
+    private LocalTime deadlineTime = null;
+
+    /**
+     * Constructor for task that does not specify whether task is done
+     * Sets done to false by default
+     * @param title name of the task
+     * @param deadline deadline of the task
+     */
+    public Deadline(String title, String deadline) {
+        super(title);
+        this.parseDate(deadline);
+    }
+
+    /**
+     * Constructor for task that specifies whether task is done
+     * Used when tasks are loaded from storage
+     *
+     * @param title name of task
+     * @param done whether task is done or not
+     * @param deadline deadline of task
+     */
+    public Deadline(String title, Boolean done, String deadline) {
```

Comment:
Boolean variables/methods can be renamed to sound like booleans. In this case, you can consider using "isDone". Same comment applies to subsequent classes of Tasks that contain this boolean variable.